container/Dockerfile.dynamo

```diff
@@ -0,0 +1,267 @@
+    PATH="/opt/dynamo/venv/bin:${PATH}"
+
+##################################
+##### Wheel Build Image ##########
+##################################
+
+# Redeclare ARCH_ALT ARG so it's available for interpolation in the FROM instruction
+ARG ARCH_ALT
+
+FROM quay.io/pypa/manylinux_2_28_${ARCH_ALT} AS wheel_builder
+
+ARG CARGO_BUILD_JOBS
+# Set CARGO_BUILD_JOBS to 16 if not provided
+# This is to prevent cargo from building $(nproc) jobs in parallel,
+# which might exceed the number of opened files limit.
+ENV CARGO_BUILD_JOBS=${CARGO_BUILD_JOBS:-16}
+# Use build arg RELEASE_BUILD = true to generate wheels for Python 3.10, 3.11 and 3.12.
+ARG RELEASE_BUILD
+
+WORKDIR /workspace
+
+RUN dnf update -y \
+    && dnf install -y llvm-toolset protobuf-compiler python3.12-devel \
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    CARGO_TARGET_DIR=/workspace/target \
+    VIRTUAL_ENV=/opt/dynamo/venv \
+    NIXL_PREFIX=/opt/nvidia/nvda_nixl
+
+COPY --from=base $RUSTUP_HOME $RUSTUP_HOME
+COPY --from=base $CARGO_HOME $CARGO_HOME
+COPY --from=base $NIXL_PREFIX $NIXL_PREFIX
+COPY --from=base $VIRTUAL_ENV $VIRTUAL_ENV
+ENV PATH=$CARGO_HOME/bin:$VIRTUAL_ENV/bin:$PATH
+
+# Copy configuration files first for better layer caching
+COPY pyproject.toml README.md LICENSE Cargo.toml Cargo.lock rust-toolchain.toml /workspace/
+
+# Copy source code
+COPY lib/ /workspace/lib/
+COPY components/ /workspace/components/
+
+# Build dynamo wheel
+RUN uv build --wheel --out-dir /workspace/dist && \
+    cd /workspace/lib/bindings/python && \
+    uv pip install maturin[patchelf] && \
+    maturin build --release --features block-manager --out /workspace/dist && \
+    if [ "$RELEASE_BUILD" = "true" ]; then \
+        # do not enable KVBM feature, ensure compatibility with lower glibc
+        uv run --python 3.11 maturin build --release --out /workspace/dist && \
+        uv run --python 3.10 maturin build --release --out /workspace/dist; \
+    fi
+
+##############################################
+########## Dev entrypoint image ##############
+##############################################
+FROM base AS dev
+
+# Application environment variables
+ENV DYNAMO_HOME=/workspace \
+    CARGO_TARGET_DIR=/workspace/target \
+    PYTHONPATH=/workspace:$PYTHONPATH
+
+WORKDIR /workspace
+
+COPY --from=wheel_builder /workspace/dist/*.whl /workspace/wheelhouse/
+COPY --from=wheel_builder $CARGO_TARGET_DIR $CARGO_TARGET_DIR
+
+# Copy Cargo cache to avoid re-downloading dependencies
+COPY --from=wheel_builder $CARGO_HOME $CARGO_HOME
+
+# Copy rest of the code
+COPY . /workspace
+
+# Common dependencies
+RUN --mount=type=bind,source=./container/deps/requirements.txt,target=/tmp/requirements.txt \
+    uv pip install --requirement /tmp/requirements.txt
+
+# Install test dependencies
+RUN --mount=type=bind,source=./container/deps/requirements.test.txt,target=/tmp/requirements.txt \
```

Comment:
Same as the other dependencies, we should move these higher in the stage if they are not going to be changing frequently